### PR TITLE
feat(affiliate): add Station affiliate config alongside vultisig-0

### PR DESCRIPTION
**Pull request by gomesalexandre**
## what

Adds Station-affiliate config siblings alongside the existing vultisig-0 configs. Zero behavior change for Vultisig Windows/iOS — those import the unchanged vultisig-0 constants.

Station consumers (mcp-ts — separate PR) will import the station* configs.

## config values

| Aggregator | Field | Value |
|---|---|---|
| THORChain + Maya | `affiliateFeeAddress` | `stvs` (live on-chain) |
| 1inch | `referrer` | `0x649E1289fD780C2F9A3D27476511283EB0d0076D` |
| KyberSwap | `referral` / `feeReceiver` | `0x649E1289fD780C2F9A3D27476511283EB0d0076D` |
| KyberSwap | `source` | `vultisig-v0` (temp fallback — Station source ID pending Kyber confirmation, fees still flow correctly) |

## injection seam

`findSwapQuote` now accepts an optional `affiliateConfig?: SwapAffiliateConfig` that carries all three per-aggregator configs and threads them through. When absent, each aggregator falls back to its vultisig-0 default — zero blast radius for existing callers.

The individual functions (`buildAffiliateParams`, `getOneInchSwapQuote`, `getKyberSwapRoute`, `getKyberSwapTx`) each get an optional config param with the same default-to-vultisig-0 pattern.

## files added

- `swap/native/stationNativeSwapAffiliateConfig.ts`
- `swap/general/oneInch/stationOneInchAffiliateConfig.ts`
- `swap/general/kyber/stationConfig.ts`
- `swap/affiliate/station.ts` (barrel re-export)

## tests

547 tests pass (544 existing unchanged + 3 new proving Station config flows through buildAffiliateParams / getOneInchSwapQuote / getKyberSwapQuote).

## receipts

```
$ curl -s https://gateway.liquify.com/chain/thorchain_api/thorchain/thorname/stvs | jq
{
  "name": "stvs",
  "expire_block_height": 56539837,
  "owner": "thor18g2rlr8845g64xr60qvljyk3n3pnf8s5w352fm",
  "preferred_asset": "ETH.USDC-0XA0B86991C6218B36C1D19D4A2E9EB0CE3606EB48",
  "aliases": [
    {"chain": "THOR", "address": "thor18g2rlr8845g64xr60qvljyk3n3pnf8s5w352fm"},
    {"chain": "ETH", "address": "0x649E1289fD780C2F9A3D27476511283EB0d0076D"}
  ]
}
```

`stvs` is live. ETH alias matches the Station EVM address in the config. Affiliate revenue accumulates as RUNE and auto-swaps to ETH USDC at threshold.

🤖 Agent-generated

## Summary by CodeRabbit

* **New Features**
  * Added station-specific affiliate configurations for multiple swap providers (native, Kyber, 1inch), enabling customizable affiliate settings per aggregator.
  * Enhanced swap quoting system to accept optional affiliate configuration overrides, providing greater control over affiliate parameters across different swap routes.

[Review Change Stack](https://app.coderabbit.ai/change-stack/vultisig/vultisig-sdk/pull/517?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: packages/core/chain/swap/general/kyber/stationConfig.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export const stationKyberSwapAffiliateConfig = {
+  // Station source ID pending Kyber partner-team confirmation.
+  // Using vultisig-v0 as a temp fallback per spec — fees still flow to
+  // Station's feeReceiver correctly; only Kyber's attribution dashboard
+  // is mis-tagged until the new source ID is registered.
```
Comment (NeOMakinG) [inline packages/core/chain/swap/general/kyber/stationConfig.ts:5]: `should-fix:` TODO needed for Kyber source ID confirmation tracking — Comment mentions 'pending Kyber partner-team confirmation' but no tracking of when this needs to be updated. Using 'vultisig-v0' as fallback means attribution is mis-tagged until confirmation arrives.

fix: Add a TODO comment like: 'TODO: Track in issue tracker when Station source ID is confirmed by Kyber. Update to correct source ID once available.'

File: packages/core/chain/swap/affiliate/station.test.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  memo: '=:e:0x86d526d6624AbC0178cF7296cD538Ecc080A95F1',
+  notes: '',
+  outbound_delay_blocks: 0,
+  outbound_delay_seconds: 0,
+  recommended_min_amount_in: '0',
+  warning: '',
+}
+
+describe('Station affiliate config', () => {
+  it('stationNativeSwapAffiliateConfig uses stvs as affiliate address', async () => {
```
Comment (NeOMakinG) [inline packages/core/chain/swap/affiliate/station.test.ts:27]: `should-fix:` Minimal test coverage for edge cases — Tests verify happy paths only — no coverage for partial affiliateConfig (e.g., only native but not oneInch), empty config, undefined config, or malformed configs. Given this is financial config affecting fee routing, adversarial test coverage would be valuable.

fix: Add test cases for: (1) partial affiliateConfig objects, (2) empty affiliateConfig, (3) undefined affiliateConfig, (4) malformed config objects (missing required fields).

File: packages/core/chain/swap/quote/findSwapQuote.ts [hunk 3]
```diff
@@ -22,15 +25,27 @@ import { getSwapAffiliateBps, VultDiscountTier } from '../affiliate'
...
+import { NativeSwapAffiliateConfig } from '../native/api/affiliate'
 import { getNativeSwapQuote } from '../native/api/getNativeSwapQuote'
 import { nativeSwapChains, nativeSwapEnabledChainsRecord } from '../native/NativeSwapChain'
 import { getNativeSwapDecimals } from '../native/utils/getNativeSwapDecimals'
 import { SwapQuote } from './SwapQuote'
 
+/** Optional per-aggregator affiliate overrides. When absent each aggregator
+ * falls back to its own vultisig-0 default — no behavior change for existing
+ * callers. Station consumers pass the station* configs from the station barrel. */
+export type SwapAffiliateConfig = {
```
Comment (NeOMakinG) [inline packages/core/chain/swap/quote/findSwapQuote.ts:38]: `suggestion:` JSDoc for affiliateConfig usage — SwapAffiliateConfig is well-designed as optional, but consider explaining that consumers should pass either all three configs or none to avoid inconsistent fee routing across aggregators.

fix: Add comment: 'Note: Consumers should pass either all three configs or none. Mixed configuration (partial override) may lead to inconsistent affiliate routing across aggregators.'

File: packages/core/chain/swap/native/stationNativeSwapAffiliateConfig.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export const stationNativeSwapAffiliateConfig = {
+  affiliateFeeAddress: 'stvs',
+  referralDiscountAffiliateFeeRateBps: 35,
```
Comment (NeOMakinG) [inline packages/core/chain/swap/native/stationNativeSwapAffiliateConfig.ts:3]: `q:` Thorname expiration monitoring — The 'stvs' thorname is hardcoded. Runtime validation that 'stvs' still resolves to expected ETH address (0x649E1289fD780C2F9A3D27476511283EB0d0076D) is missing. Thornames expire (expire_block_height in receipt).

fix: Consider adding thorname validation step or monitoring for expiration.

File: packages/sdk/src/index.ts [hunk 5]
```diff
@@ -194,6 +194,14 @@ export {
 export type { SwapKitConfig } from '@vultisig/core-chain/swap/general/swapkit/config'
 export { configureSwapKit, getSwapKitConfig } from '@vultisig/core-chain/swap/general/swapkit/config'
 
+// Station affiliate configs — parallel to vultisig-0 defaults, for Station consumers
```
Comment (NeOMakinG) [inline packages/sdk/src/index.ts:197]: `q:` Documentation for config usage guidance — SDK exports station configs alongside vultisig-0 defaults. Documentation needed on when consumers should use which config set (vultisig-0 for Windows/iOS, station* for Station-affiliated consumers).

fix: Consider adding documentation or comments explaining intended use cases.

**Review by NeOMakinG — APPROVED** [review]
gm — multi-lane pass: . runtime-tested on tier1-sdk-library. lgtm.

## testing approach
- tier1-sdk-library

## should-fix
### 1. TODO needed for Kyber source ID confirmation tracking — `packages/core/chain/swap/general/kyber/stationConfig.ts:5`
Comment mentions 'pending Kyber partner-team confirmation' but no tracking of when this needs to be updated. Using 'vultisig-v0' as fallback means attribution is mis-tagged until confirmation arrives.
action: Add a TODO comment like: 'TODO: Track in issue tracker when Station source ID is confirmed by Kyber. Update to correct source ID once available.'

### 2. Minimal test coverage for edge cases — `packages/core/chain/swap/affiliate/station.test.ts:27`
Tests verify happy paths only — no coverage for partial affiliateConfig (e.g., only native but not oneInch), empty config, undefined config, or malformed configs. Given this is financial config affecting fee routing, adversarial test coverage would be valuable.
action: Add test cases for: (1) partial affiliateConfig objects, (2) empty affiliateConfig, (3) undefined affiliateConfig, (4) malformed config objects (missing required fields).

## suggestion
### 1. JSDoc for affiliateConfig usage — `packages/core/chain/swap/quote/findSwapQuote.ts:38`
SwapAffiliateConfig is well-designed as optional, but consider explaining that consumers should pass either all three configs or none to avoid inconsistent fee routing across aggregators.
action: Add comment: 'Note: Consumers should pass either all three configs or none. Mixed configuration (partial override) may lead to inconsistent affiliate routing across aggregators.'

### 2. Consider validation helper for affiliateConfig — `packages/sdk/src/tools/swap/findSwapQuote.ts:69`
SDK wrapper correctly threads affiliateConfig through, but validation (address format, bps ranges, thorname resolution) before reaching swap providers would provide earlier/clearer errors to consumers.
action: Add a validateSwapAffiliateConfig helper that checks: EVM addresses are valid, thorname 'stvs' resolves, bps values within valid ranges (0-10000).

## q
- Thorname expiration monitoring
- Documentation for config usage guidance

## risk
Low risk. Optional parameters with sensible defaults mean zero blast radius for callers not passing affiliateConfig. New configs properly exported and typed. Main risks: (1) Station thorname 'stvs' could expire (no runtime validation), (2) Kyber source ID 'vultisig-v0' is temporary fallback pending confirmation, (3) partial affiliateConfig usage could lead to inconsistent routing.

## verdict
APPROVE — ship it 🫡

🤖 reviewed via [Claude Code](https://claude.com/claude-code) with codex `gpt-5.3-codex` adversarial pass. tier1-sdk-library.
